plugins/hls-semantic-tokens-plugin/src/Ide/Plugin/SemanticTokens/Internal.hs

```diff
@@ -60,23 +58,24 @@ import           Language.LSP.Protocol.Types          (NormalizedFilePath,
                                                        type (|?) (InL))
 import           Prelude                              hiding (span)
 
-logActionWith :: (MonadIO m) => IdeState -> Priority -> String -> m ()
-logActionWith st prior = liftIO . logPriority (ideLogger st) prior . T.pack
 
 -----------------------
 ---- the api
 -----------------------
 
-computeSemanticTokens :: IdeState -> NormalizedFilePath -> ExceptT PluginError Action SemanticTokens
-computeSemanticTokens st nfp = do
-  logActionWith st Debug $ "Computing semantic tokens:" <> show nfp
+computeSemanticTokens :: Recorder (WithPriority SemanticLog) -> PluginId -> IdeState -> NormalizedFilePath -> ExceptT PluginError Action SemanticTokens
+computeSemanticTokens recorder pid _ nfp = do
+  logWith recorder Debug (LogMsg "computeSemanticTokens start")
+  config :: SemanticTokensConfig  <- lift $ usePropertyAction #tokenMapping pid semanticConfigProperties
+  logWith recorder Debug (LogMsg $ show $ toJSON config)
+  logWith recorder Debug (LogConfig config)
```

Comment:
Surely you don't need both of these?

Reply:
The ToJason would be deleted after we finalize the choice of configuration.

Reply:
done